### DISCUSSION TITLE
[cloud_firestore_odm_generator] Support Firestore Timestamps

### DISCUSSION THREAD

**Discussion by FXschwartz** [firebase/flutterfire#8029] [Feature request]
Currently when running `flutter pub run build_runner build --delete-conflicting-outputs` against a model that contains a timestamp throws an error.

```
Could not generate `fromJson` code for `pressedOn`.
To support the type `Timestamp` you can:
* Use `JsonConverter`
  https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonConverter-class.html
* Use `JsonKey` fields `fromJson` and `toJson`
  https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/fromJson.html
  https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/toJson.html
package:buttonSimple/models/button_event.dart:20:19
```

Looks like this is on json_serializable to implement but it might help to have an issue here to track for people running into this and assuming it's a bug.

Currently, there is an open[ enhancement](https://github.com/google/json_serializable.dart/issues/1072) in the json_serializable repo

**Comment by inamit** [comment]
Is there a workaround for this issue for now? I have tried making the field a `DateTime` and implementing a `JsonConverter`, but `flutter pub run build_runner build --delete-conflicting-outputs` doesn't generate a `where` or `orderby` functions for this field.

**Reply by FXschwartz** [reply]
I haven't found a fully working workaround yet, all of my projects have `DateTime` fields so I, unfortunately, can't use ODM until this gets implemented.
